### DISCUSSION TITLE
how to connect to AVR

### DISCUSSION THREAD

**Discussion by Kimdaegeun1230** [olikraus/u8g2#1510] [Q&A]
Now I have ST7920 and Atmega128,
but I don't know how to connect to them.
please tell me something...
which pin should I connect to ST7920?

**Comment by olikraus** [comment]
What's the question? How to wire the display or how to setup the atmega?

**Reply by Kimdaegeun1230** [reply]
I want to know both of them.
and I use MicroChip studio with C language  now.
![GLCD pin](https://user-images.githubusercontent.com/69760411/120785952-a32f7880-c568-11eb-9e13-01a94ec4ee78.jpg)
Thank you for your help.

**Reply by olikraus** [reply]
For the display you need to consult the datasheet of the display, but your specific display has been discussed here: https://github.com/olikraus/u8g2/wiki/setup_tutorial#identify-digital-lines
There should be also several youtube videos regarding the ST7920 and u8g2.

MicroChip Studio is more complicated. Basically u8g2 is an Arduino Library, so anything else is not supported. However there is a porting guide:
https://github.com/olikraus/u8g2/wiki/Porting-to-new-MCU-platform

More over there is also a guide with Atmel Studio (not sure whether this is similar to MicroChip Studio):
https://github.com/olikraus/u8g2/wiki/u8g2as7

**Reply by Kimdaegeun1230** [reply]
https://www.instructables.com/ST7920-LCD-With-ATmega328-in-Atmel-Studio-Using-SP/
Connect the pin as written in this link. 
I did like this. 
 
PIN Argument in main.c		
PIN_Fuction /             (LCD)    /    (ATMEGA128)
CLK_PIN / 			E / 		      SCK	
DATA_PIN / 			RW / 	    MOSI
CS_PIN / 		 		RS / 	       SS
DC_PIN / 		 
RESET_PIN / 			RST /              CLK
 
But I don't know what kind of pin DC_PIN is used in LCD and Atmega.

**Reply by olikraus** [reply]
As far as I remember the CS pin is not required for your display. 
Also do not confuse CS with the RS pin.
RS and DC are just two different names for the same pin (as mentioned here: https://github.com/olikraus/u8g2/wiki/u8g2setupcpp#wiring)

> But I don't know what kind of pin DC_PIN is used in LCD and Atmega.

DC_PIN is the pin labeled with RS on your display.
DC_PIN is just a normal digital pin, so any digital output of the ATMega will be ok.
